### PR TITLE
Fix Arrays with leading nulls

### PR DESCRIPTION
**Pull request by eeeebbbbrrrr**
This test fails with some form of:

```
compression method lz4 not supported
Postgres location: toast_compression.c:187
```

A classic off-by-one error caused this:
If element 0 was null, and 1 was non-null, data started with 1,
but iteration would advance before unpacking.
This allowed moving to uninit data, or just skipping elements.

### CODE REVIEW

File: pgrx-tests/src/tests/array_tests.rs [hunk 1]
```diff
@@ -482,4 +482,16 @@ mod tests {
...
+
+    #[pg_test]
+    fn test_text_array() -> Result<(), Box<dyn std::error::Error>> {
+        let a = Spi::get_one::<Array<String>>(
+            "SELECT ARRAY[NULL, NULL, NULL, NULL, 'the fifth element']::text[]",
+        )?
+        .expect("spi result was NULL")
+        .into_iter()
+        .collect::<Vec<_>>();
+        assert_eq!(a, vec![None, None, None, None, Some(String::from("the fifth element"))]);
```
Comment (BradyBonnette) [inline pgrx-tests/src/tests/array_tests.rs:494]: If this passes on all distros in every PG configuration, does that mean it is..

... multi-pass?

Comment (eeeebbbbrrrr) [reply pgrx-tests/src/tests/array_tests.rs:494]: when it does, YES!  hahaha

**Comment by eeeebbbbrrrr** [comment]
A different failure, but probably the same?

```
attempt to subtract with overflow
Postgres location: varlena.rs:236
```

**Comment by workingjubilee** [comment]
No idea why it is specifically failing to link and run the versioned so tests.

**Review by workingjubilee — APPROVED** [review]
I assert that the remaining test failures are unrelated.

**Pull request merged**